auth0/src/main/java/com/auth0/android/authentication/storage/SecureCredentialsManager.java

```diff
@@ -310,4 +361,42 @@ long getCurrentTimeInMillis() {
         return clock.getCurrentTimeMillis();
     }
 
+    private boolean hasScopeChanged(@NonNull String storedScope, @Nullable String requiredScope) {
+        if (requiredScope == null) {
+            return false;
+        }
+        String[] stored = storedScope.split(" ");
+        Arrays.sort(stored);
+        String[] required = requiredScope.split(" ");
+        Arrays.sort(required);
+        return stored != required;
```

Comment:
Use `!Arrays.equals(stored, required);`, as `!=` will compare the references and not the contents.